### ISSUE TITLE
Remove all todo tests and file dedicated tickets for them

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7776]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#7776|
|Type|enhancement|
|Created|2015-02-07 14:13:47Z|
|Branch|https://github.com/twisted/twisted/tree/todo-removal-7776|

TODO tests are useless. We should remove them from code and file dedicated ticket for each todo test (or one ticket for related tests)

------

From mailing lists

```
The default state of a test should *not* be to skip.  It should be to run.
A skipped test is a useless test.

As a matter of general principle, the only reason a test should ever be
skipped within Twisted is if an optional dependency is not available.  And,
just to be clear, the presence of "TODO'd" tests is pretty much just a bug,
fix old ones and never add new ones :-).

I think that these todo are just another way of creating a ticket ...
with a promise to annoy you forever

And to annoy every contributor ever who works on something unrelated :).
```

<details><summary>Searchable metadata</summary>

```
trac-id__7776 7776
type__enhancement enhancement
reporter__adiroiban adiroiban
priority__normal normal
milestone__ 
branch__branches_todo_removal_7776 branches/todo-removal-7776
branch_author__adiroiban adiroiban
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1423318427994962 1423318427994962
changetime__1426524713417421 1426524713417421
version__None None
owner__adiroiban adiroiban

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_1"></a>|@adiroiban commented|
|-|-|

(In [43874]) Branching to todo-removal-7776.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban removed owner|
|-|-|


Follow up tickets [#7777](https://github.com/twisted/twisted/issues/7777) [#7778](https://github.com/twisted/twisted/issues/7778) [#6718](https://github.com/twisted/twisted/issues/6718) [#7779](https://github.com/twisted/twisted/issues/7779) [#7781](https://github.com/twisted/twisted/issues/7781) [#7782](https://github.com/twisted/twisted/issues/7782) [#7783](https://github.com/twisted/twisted/issues/7783) [#7784](https://github.com/twisted/twisted/issues/7784) [#7785](https://github.com/twisted/twisted/issues/7785)

Does twisted still support linux 2.2 ? :) I have just removed a todo test for linux 2.2

I have converted conditionals todo into skips and made sure that there is a ticket for them.


I have searched the code for `.todo` pattern.

I have removed FIXME references to [#6008](https://github.com/twisted/twisted/issues/6008) as twisted code should no longer have todo markers.

I triggered the buildbot and checked that there are no more expected failures.


-----

As a follow up, maybe we should update the docs to mention that todo tests should not be merged in master...or at least add this to the reviewer's check list.


Please check that changes make sense. Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph commented|
|-|-|

Twisted does not support Linux 2.2.  Hopefully nobody is building new devices or installing new OSes that require it :-).

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph set owner to @adiroiban|
|-|-|

This looks great, thanks for filing all those tickets.  Land, please!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban commented|
|-|-|

Thanks for the review. I have created [#7819](https://github.com/twisted/twisted/issues/7819) as a follow up of this so that in the future we don't merge todo tests.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_6"></a>|@adiroiban set status to `closed`|
|-|-|

(In [44066]) Merging todo-removal-7776: Remove all TODO tests.

Author: adiroiban
Reviewer: glyph
Fixes: [#7776](https://github.com/twisted/twisted/issues/7776)
